dining/views.py

```diff
@@ -0,0 +1,348 @@
+from datetime import datetime, timedelta
+
+from dining.constants import map_to_pretty
+from django.db.models import Sum
+from django.http import HttpResponseBadRequest, HttpResponseForbidden, JsonResponse
+from legacy.models import Account, DiningBalance, DiningTransaction
+from options.models import Option, get_option, get_value
+from pytz import timezone
+from rest_framework.views import APIView
+
+from studentlife.utils import date_iso_eastern, get_new_start_end
+
+
+class Dashboard(APIView):
+    def get(self, request, format=None):
+        if not self.request.user.is_authenticated:
+            return HttpResponseForbidden()
+
+        pennid = request.user.username
+
+        try:
+            uid = Account.objects.get(pennid=pennid).id
+        except Account.DoesNotExist:
+            return HttpResponseBadRequest()
+
+        json = self.balance(uid)
+
+        start, end = self.get_semester_start_end()
+
+        eastern = timezone("US/Eastern")
+        start = start.replace(tzinfo=eastern, hour=0, minute=0, second=0, microsecond=0)
+        end = end.replace(tzinfo=eastern, hour=0, minute=0, second=0, microsecond=0)
+        json["start-of-semester"] = date_iso_eastern(start)
+        json["end-of-semester"] = date_iso_eastern(end)
+
+        json["cards"] = {"recent-transactions": self.recent_transactions_card(uid)}
+
+        average_balances = self.get_average_balances(uid, start)
+
+        if average_balances:
+            json["cards"].update({"daily-average": average_balances})
+
+        dollar_prediction = self.get_prediction_dollars(uid, start, end)
+
+        if dollar_prediction:
+            json["cards"].update({"predictions-graph-dollars": dollar_prediction})
+
+        swipes_prediction = self.get_prediction_swipes(uid, start, end)
+
+        if swipes_prediction:
+            json["cards"].update({"predictions-graph-swipes": swipes_prediction})
+
+        frequent_locations = self.get_frequent_locations(uid, start, end)
```

Comment:
```suggestion
        update_if_not_none(json["cards"], self.get_frequent_locations(uid, start, end))
```
I feel like this could clean up a lot of repeated code in this function and make it even easier to read if you factor out the `if` check to a helper function.